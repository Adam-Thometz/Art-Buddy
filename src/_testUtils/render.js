import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
