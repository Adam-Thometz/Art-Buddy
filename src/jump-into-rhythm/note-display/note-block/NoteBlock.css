.NoteBlock {
  border: 10px solid;
  height: 15vh;
  width: 10vw;
  margin-left: 8%;
  border-radius: 25px;
}

.NoteBlock img {
  margin-top: 20px;
}