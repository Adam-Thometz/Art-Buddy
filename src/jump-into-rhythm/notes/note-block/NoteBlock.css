.NoteBlock {
  border: 10px solid;
  height: 15vh;
  width: 10vw;
  margin-left: 8%;
  border-radius: 25px;
  position: relative;
}

.NoteBlock-note {
  margin-top: 20px;
}

.NoteBlock-delete {
  height: 24px;
  width: 24px;
  position: absolute;
  top: 12px;
  right: 12px;
  cursor: pointer;
}