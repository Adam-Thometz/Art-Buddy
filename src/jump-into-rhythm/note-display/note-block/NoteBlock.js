import React from 'react';

import './NoteBlock.css';

const NoteBlock = ({ borderColor }) => {
  return (
    <div className='NoteBlock' style={{ borderColor }}>

    </div>
  );
};

export default NoteBlock;