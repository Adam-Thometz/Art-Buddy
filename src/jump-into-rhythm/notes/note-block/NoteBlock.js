import { useDispatch } from "react-redux";
import { deleteFromRhythm } from "_redux/jump-into-rhythm/jumpIntoRhythmActions";

import "./NoteBlock.css";

import deleteBtn from "_media/_general/close.png";

const NoteBlock = ({ borderColor, note, id }) => {
  const dispatch = useDispatch();

  const handleDeleteNote = () => dispatch(deleteFromRhythm(id));

  return (
    <div className="NoteBlock" style={{ borderColor }} data-testid="block">
      {note ? (
        <>
          <img
            src={deleteBtn}
            className="NoteBlock-delete"
            alt="Delete note"
            onClick={handleDeleteNote}
          />
          <img src={note.img} className="NoteBlock-note" alt="" />
        </>
      ) : null}
    </div>
  );
};

export default NoteBlock;
