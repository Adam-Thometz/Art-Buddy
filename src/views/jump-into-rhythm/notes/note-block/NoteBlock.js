import { useDispatch } from "react-redux";
import { deleteFromRhythm } from "store/jump-into-rhythm/jumpIntoRhythmReducer";

import "./NoteBlock.css";

import { CLOSE } from "assets/_general/general.assets";

const NoteBlock = ({ borderColor, note, id }) => {
  const dispatch = useDispatch();

  const handleDeleteNote = () => dispatch(deleteFromRhythm(id));

  return (
    <div className="NoteBlock" style={{ borderColor }} data-testid="block">
      {note ? (
        <>
          <img
            src={CLOSE}
            className="NoteBlock-delete"
            alt="Delete note"
            onClick={handleDeleteNote}
          />
          <img src={note.img} className="NoteBlock-note" alt="" />
        </>
      ) : null}
    </div>
  );
};

export default NoteBlock;
