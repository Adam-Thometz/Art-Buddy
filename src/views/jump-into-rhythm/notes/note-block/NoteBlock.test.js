import NoteBlock from "./NoteBlock";

import { render } from "test/render";

describe("NoteBlock component", () => {
  it("renders without crashing", () => {
    render(<NoteBlock />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<NoteBlock />);
    expect(asFragment()).toMatchSnapshot();
  });
});
