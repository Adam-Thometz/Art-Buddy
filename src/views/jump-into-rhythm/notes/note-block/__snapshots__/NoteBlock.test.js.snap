// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteBlock component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NoteBlock"
    data-testid="block"
  />
</DocumentFragment>
`;
